### PR TITLE
Spec 161 miden deployment fix from package

### PR DESCRIPTION
**Pull request by sshrihar**
Update the deployment workflow. Updates
- Single workflow for deploying to testnet and mainnet instances
- Deploys to new instance created using terraform
- devnet builds on push to next branch. It builds a new package and uses that to deploy on testnet.
- When built from released or prerelease condition the deployment occurs on new testnet instance

### CODE REVIEW

**Review by djpolygon — APPROVED** [review]

**Review by Mirko-von-Leipzig — COMMENTED** [review]
This looks like it should work thanks!

What do you think of merging the implementations here with the workflow/outline of #473? The [release](https://github.com/0xPolygonMiden/miden-node/pull/473/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34) flow should given a good indication of the intent; the devnet one I left in a fiddly state but should be much simpler as its only arm build & deploy.

This would require:

- Moving the deploy code into an action (it always uses a local package file and deploys via the S3 bucket). This deduplicates this code.
- Moving package building into an action. The arch is chosen simply the caller. __I assume that package building is the same for both amd/arm__?.

File: .github/actions/ssm_execute/action.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+              --query "CommandInvocations[0].Status" \
+              --output text)
+          echo "Command Status: $STATUS"
+
+          OUTPUT=$(aws ssm list-command-invocations \
+              --command-id ${{ steps.execute_cmd.outputs.command_id }} \
+              --details \
+              --query "CommandInvocations[0].CommandPlugins[0].Output" \
+              --output text)
+          echo "Command Output: $OUTPUT"
```
[the remark this answers is not in the document]
Comment (sshrihar) [reply .github/actions/ssm_execute/action.yml:60]: It does print output on checking every 10 seconds where there is something to print.

File: .github/workflows/deploy_package.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+          else
+            echo "package_postfix=arm64.deb" >> $GITHUB_ENV
+          fi
+
+      - name: Stop miden services
+        uses: ./.github/actions/ssm_execute
+        with:
+          instance_id: ${{ secrets[inputs.instance_id] }}
+          command: |
+            sudo rm -f miden-*; sudo systemctl stop miden-node; sudo systemctl stop miden-faucet; sudo apt remove miden-node miden-faucet -y
```
[the remark this answers is not in the document]
Comment (sshrihar) [reply .github/workflows/deploy_package.yml:68]: We can extend the code as per your suggestion in a test branch.

File: .github/actions/ssm_execute/action.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mirko-von-Leipzig) [reply .github/actions/ssm_execute/action.yml:60]: Yes but is there ever anything to print? My experience was that there is only output available when the entire set of commands is complete.

**Comment by sshrihar** [comment]
hi @Mirko-von-Leipzig , can we merge these changes and continue to work on the updates in a new branch.

**Pull request merged**